### PR TITLE
add paiv/uxn-ios

### PR DESCRIPTION
**Pull request by craigcomstock**
Last commit July 2021.
I noticed they didn't include uxn as a submodule, so I could fork and fix that, seems like a better idea right? I will go through and review how complete it is and update this entry.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,6 +51,7 @@ The [Uxn](https://100r.co/site/uxn.html) ecosystem is a personal computing playg
   - [Raspberry Pi Pico](https://git.sr.ht/~alderwick/pico-uxn) - Partial
   - [Teletype](https://github.com/csboling/teluxn) - Incomplete.
   - [ESP32](https://github.com/max22-/uxn-esp32) - Incomplete.
+  - [iOS](https://github.com/paiv/uxn-ios) - Incomplete.
```
Comment (craigcomstock) [inline README.md:54]: Ok, I would say, yes, based on the code here: https://github.com/paiv/uxn-ios/blob/cde481553be8e559a91c6b4d30f3ffb2de9ac8e8/app/Emulator/Platform/app_uxn.c#L163-L178

```
    devsystem = portuxn(u, 0x0, "system", system_talk);
    portuxn(u, 0x1, "console", console_talk);
    devscreen = portuxn(u, 0x2, "screen", screen_talk);
    devaudio0 = portuxn(u, 0x3, "audio0", audio_talk);
    portuxn(u, 0x4, "audio1", audio_talk);
    portuxn(u, 0x5, "audio2", audio_talk);
    portuxn(u, 0x6, "audio3", audio_talk);
    portuxn(u, 0x7, "---", nil_talk);
    portuxn(u, 0x8, "controller", nil_talk);
    devmouse = portuxn(u, 0x9, "mouse", nil_talk);
    portuxn(u, 0xa, "file", file_talk);
    portuxn(u, 0xb, "datetime", datetime_talk);
    portuxn(u, 0xc, "---", nil_talk);
    portuxn(u, 0xd, "---", nil_talk);
    portuxn(u, 0xe, "---", nil_talk);
    portuxn(u, 0xf, "---", nil_talk);
```

That this is incomplete, though more complete than the other iOS project.

The app comes bundled with a lot of ROMs, all of which seems to work, like piano.

Comment (craigcomstock) [inline README.md:54]:
```suggestion
  - [iOS](https://github.com/paiv/uxn-ios) - Nearly Complete (missing midi and controller).
```

**Comment by craigcomstock** [comment]
@neauoire this port looks pretty good. I'd be happy to fork it and improve it as needed and submit those upstream and see if the original author is interested. Eventually I'd like to make this iOS app where I nudge in my natural'char language using varvara :) Cheers!

File: README.md (hunk 1 above)
Comment (craigcomstock) [reply README.md]: @neauoire what about the mouse business?

**Comment by neauoire** [comment]
Thanks for the PR! There has been quite a few people wanting to run roms on their iOS devices on the Lines forum in the past, it'd be nice if you could bring it up to specs :)

**Comment by craigcomstock** [comment]
> Thanks for the PR! There has been quite a few people wanting to run roms on their iOS devices on the Lines forum in the past, it'd be nice if you could bring it up to specs :)

Cool. Will do that work (slowly most likely). I am getting a usb camera adapter today so I can plugin a keyboard, mouse, midi adapter to my iphone 6. Will see what I can nudge!

**Pull request merged**